can/io/logger.py

```diff
@@ -225,6 +226,11 @@ def _get_new_writer(self, filename: StringPathLike) -> FileIOMessageWriter:
             return logger
         elif isinstance(logger, Printer) and logger.file is not None:
             return cast(FileIOMessageWriter, logger)
+        elif isinstance(logger, can.io.SqliteWriter):
+            raise Exception(
+                "The SqliteWriter is not available in conjunction"
+                "with the sized rotating logger."
+            )
         else:
             raise Exception(
                 "The Logger corresponding to the arguments is not a FileIOMessageWriter or "
```

Comment:
You could improve the existing error message without the `isinstance(logger, can.io.SqliteWriter)` check. You can access the logger class name with `logger.__class__.__name__`, for example inside a f-string.
This way you do not need special cases for any new unsupported loggers